### DISCUSSION TITLE
How many ping failures does it take to google to trigger a no Internet event?

### DISCUSSION THREAD

**Discussion by ppgriffith** [InternetConnectionMonitor/icm-public#99] [Q&A] [answered]
I'm using ICM on my own home internet and on several of my clients servers (I work in IT). I'm getting a lot of Internet "outages" reported by ICM, without actually having an actual outage. Is one ping fail enough to trigger it or is it several in a row?
I got a 2 second outage an hour or so ago, there's no way my internet could go down and reconnect in 2 seconds. Here is a log from a client of mine, crazy consistent times, has to be wrong.
![Screenshot 2024-06-17 191736](https://github.com/InternetConnectionMonitor/icm-public/assets/128910513/0eb71174-50ad-4657-9d1a-9f8a895ff83d)

**Comment by dmishh** [comment, marked as answer]
@ppgriffith yeah, that looks a bit weird. The algorithm itself is quite complex, there are different numbers for each time checking setting. There should be several unsuccessful attempts for ICM to record the connection issue. If there is an outage that could be detected by the browser - the offline event will be triggered automatically
